syftlib/src/main/java/org/openmined/syft/network/SignallingClient.kt

```diff
@@ -0,0 +1,85 @@
+package org.openmined.syft.network
+
+import io.reactivex.Flowable
+import io.reactivex.processors.PublishProcessor
+import kotlinx.serialization.json.JsonPrimitive
+import kotlinx.serialization.json.json
+import okhttp3.*
+import java.util.concurrent.TimeUnit
+
+private const val TAG = "WebRTC Signalling Client"
+
+internal class SignallingClient(
+    private val workerId: String,
+    private val keepAliveTimeout: Int = 20000,
+    private val url: String,
+    private val port: Int
+) {
+    private lateinit var request: Request
+    private lateinit var client: OkHttpClient
+    private var isConnected = false
+    private lateinit var webSocket: WebSocket
+    private val syftSocketListener = SyftSocketListener()
+
+    private val statusPublishProcessor: PublishProcessor<String> = PublishProcessor.create<String>()
+
+    fun start(): Flowable<String> {
+        client = OkHttpClient.Builder()
+            .pingInterval(keepAliveTimeout.toLong(), TimeUnit.MILLISECONDS)
+            .build()
+        request = Request.Builder()
+            .url(url)
+            .build()
+        connect()
+        return statusPublishProcessor.onBackpressureBuffer()
+    }
+
+    inner class SyftSocketListener : WebSocketListener() {
+
+        override fun onOpen(webSocket: WebSocket, response: Response) {
+            super.onOpen(webSocket, response)
+            this@SignallingClient.webSocket = webSocket
+            isConnected = true
+            statusPublishProcessor.offer("We are open for business!")
+        }
+
+        override fun onMessage(webSocket: WebSocket, text: String) {
+            statusPublishProcessor.offer("Received new message $text")
+        }
+
+        override fun onFailure(webSocket: WebSocket, t: Throwable, response: Response?) {
+            super.onFailure(webSocket, t, response)
+            isConnected = false
```

Comment:
I think we dont need to maintain isConnected anymore. First of all it faces concurrency issues if the observed thread is running late. We can close the socket without checking. Same for sending. And rather use the response of those methods for statusPublishProcessor